cmake/Dependencies.cmake

```diff
@@ -1094,6 +1094,14 @@ if(USE_NCCL)
   endif()
 endif()
 
+# ---[ XCCL
+if(USE_XCCL)
```

Comment:
Why is this post-processing and not done during XCCL detection? or when we skip detection because the flag is set?

Reply:
Currently, XCCL detection resides in torch-xpu-ops. If `USE_XPU` is set to `OFF` (e.g., due to an incomplete toolchain), torch-xpu-ops will not be included in the build (refer to https://github.com/pytorch/pytorch/blob/cf6e5d18818da43c0a239fcbba8d664cbf41a7ef/caffe2/CMakeLists.txt#L1102-L1152), leaving no opportunity to update the `USE_XCCL` flag accordingly.